### ISSUE TITLE
Renaming asset folder deletes all assets inside

### ISSUE THREAD

**Issue by dominikradl** [statamic/cms#5064]
### Bug description

When renaming an asset folder, all the assets including metadata get lost. As this is a fairly new feature, it may be, that only S3 asset containers show this behavior.

### How to reproduce

- Create an asset folder inside an S3 container
- Upload a file
- Rename the asset folder

First the filelist doesn't even show the renamed folder until after clearing the cache, then the renamed asset folder is empty in the list and in the S3 bucket.

### Versions

Statamic 3.2.30 Pro
Laravel 8.60.0
PHP 7.4.27

### Installation

Fresh statamic/statamic site via CLI

### Additional details

This is the pull request that introduced this feature: https://github.com/statamic/cms/pull/4028

_No response_

**Comment by jackmcdade** [comment]
Bumping this high priority.

**Comment by dominikradl** [comment]
Hi, are there any updates or ETA on this one?

**Comment by jackmcdade** [comment]
Actively working on it!

**Comment by arthurperton** [comment]
I'm trying to reproduce this one, but I'm not losing any assets or meta data. Could someone help me with the steps? I'm doing this:

1. In an empty S3 container, create a folder called `sub1`(and see the `sub1` folder on s3)
2. Upload an image into the newly created `sub1` folder (and see the image and the meta in the `sub1` folder on s3)
3. Rename the `sub1` folder to `sub2` (and see the image and the meta in the `sub2` folder on s3)

When I disable the file watcher, I do have to clear the cache to see the renamed folder in the Control Panel. However the image and the meta file still do not get lost, just the same result as described in the steps above.

**Comment by jackmcdade** [comment]
We've found it only happens if the folder or subfolder is named `/0/`. Give that a try.

**Comment by arthurperton** [comment]
Thanks Jack, yes actually I was the one who found that out when looking into #5405 😄 

I am curious if Jason's fix also fixes this issue. I'd have to be able to reproduce it first though. Hard to imagine that @dominikradl and @jasonvarga both used a folder called `0` in this issue too. Did you guys? Could you help me reproduce the issue or maybe test if #5433 fixes this too?

**Comment by dominikradl** [comment]
No, I didn't use a folder named `0` and still lost all the metadata. I will take a look at it right now.

**Comment by dominikradl** [comment]
I can reproduce losing the metadata (see attached video) and was also able to reproduce losing all assets inside the folder, but unfortunately didn't film it and can't reproduce it since. None of my folder is or was called `0`.

https://user-images.githubusercontent.com/84016239/157878424-b0f2ad20-ef69-4ffc-8c4c-01b298990c45.mov

**Comment by dominikradl** [comment]
Now I was able to reproduce it one more time (sorry for the long video, I assume it's benefitial for you guys to see the whole process). This time I noticed something strange but most likely related:

- First I renamed the new folder from `test` to `test100` and the asset got lost.
- Then I uploaded an asset and moved it inside `test`. When I renamed `test` to `test65` it got lost again.
- Then I created a new folder called `newfolder`. When I tried to rename it to `newestfolder`, it was only called `newest` but the asset was still there. When I then renamed it to `newestfolder`, the name was ok, but the asset was lost.

Two strange things I noticed are:

1. The new name was wrong
2. There seem to be differences between submitting the renaming by return key and by click on the button.

https://user-images.githubusercontent.com/84016239/157881141-e97ce947-642e-4030-9208-a96564ea7e35.mov

**Comment by jackmcdade** [comment]
Are the files actually lost from the filesystem, or just not showing the control panel?

**Comment by dominikradl** [comment]
Yes, they are immediately deleted from S3.

**Comment by jasonvarga** [comment]
We did fix a handful of these issues recently but it hasn't been tagged yet. Thanks for the video, I'll try to reproduce it and see if the recent fixes fix this too.

**Comment by jasonvarga** [comment]
There's definitely weirdness going on with how the control panel is showing stuff, but can you 100% confirm that the assets are actually deleted from your S3 dashboard?

**Comment by dominikradl** [comment]
Yeah, pretty weird. Video attached showing that assets are missing from S3. FYI that was renaming using return key again.

https://user-images.githubusercontent.com/84016239/157911781-e1c31b4f-f88f-411b-99c2-c904ed455d4d.mov

**Comment by jasonvarga** [comment]
So weird. But I believe you now! 😂

**Comment by jasonvarga** [comment]
What version of Statamic are you running?

**Comment by jasonvarga** [comment]
Can you find out the version of the s3 adapter by running `composer show league/flysystem-aws-s3-v3`?

**Comment by dominikradl** [comment]
```
name     : league/flysystem-aws-s3-v3
descrip. : Flysystem adapter for the AWS S3 SDK v3.x
keywords : 
versions : * 1.0.29
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage : 
source   : [git] https://github.com/thephpleague/flysystem-aws-s3-v3.git 4e25cc0582a36a786c31115e419c6e40498f6972
dist     : [zip] https://api.github.com/repos/thephpleague/flysystem-aws-s3-v3/zipball/4e25cc0582a36a786c31115e419c6e40498f6972 4e25cc0582a36a786c31115e419c6e40498f6972
path     : /Users/dominikradl/Documents/Server/brucha/vendor/league/flysystem-aws-s3-v3
names    : league/flysystem-aws-s3-v3

support
issues : https://github.com/thephpleague/flysystem-aws-s3-v3/issues
source : https://github.com/thephpleague/flysystem-aws-s3-v3/tree/1.0.29

autoload
psr-4
League\Flysystem\AwsS3v3\ => src/

requires
aws/aws-sdk-php ^3.20.0
league/flysystem ^1.0.40
php >=5.5.0

requires (dev)
henrikbjorn/phpspec-code-coverage ~1.0.1
phpspec/phpspec ^2.0.0
```

**Comment by jasonvarga** [comment]
This appears to be fixed by #5433
